developers.adoc

```diff
@@ -225,25 +225,37 @@ to `X.Y.Z` format (e.g. `0.9.0` for 0.9 release)
 
 === Python
 
-* Create a `~/.pypirc` file with this (https://github.com/pypa/setuptools/issues/941#issuecomment-279123202[source]):
+* Make sure that the needed Python packaging tools are installed and up-to-date:
+** Debian/Ubuntu/etc.: `sudo apt install python3-setuptools python3-wheel twine`
+** Other Linuxes: see https://packaging.python.org/guides/installing-using-linux-tools/[Python Packaging User Guide]
+** Everywhere else: use your system package manager, or `python3 -m pip install --upgrade setuptools twine wheel`
+* Pump version in `kaitaistruct.py`, seek `\\__version__ =`
+* Build distributions and upload them to PyPI:
 +
-[source,ini]
+[source,bash]
 ----
-[distutils]
-index-servers =
-    pypi
-
-[pypi]
-repository: https://pypi.python.org/pypi
-username: <login>
-password: <pass>
+# Delete any old previously built distributions.
+$ rm -r dist/
+# Build source and wheel distributions.
+$ python3 setup.py sdist bdist_wheel
+# Run basic correctness checks on the built distributions.
+$ twine check dist/*
+# Upload the built distributions.
+# You may want to upload to TestPyPI first as a test before uploading to the real PyPI.
+# `twine upload` will prompt for credentials,
+# or you can provide them via the environment variables TWINE_USERNAME and TWINE_PASSWORD
+# (see https://twine.readthedocs.io/en/latest/#environment-variables).
+$ twine upload --repository=testpypi dist/* # Upload to TestPyPI (https://test.pypi.org/)
+$ twine upload dist/* # Upload to live PyPI (https://pypi.org/)
```

Comment:
Yeah, I really think that these comments are unnecessary and redundant.

And please keep the style of the document, i.e. not this long shell snippets, but one command per list item, optionally having a short concise comment next to it (but only when it's really necessary, not obvious or counter-intuitive). Like this:

 - `python3 setup.py sdist bdist_wheel`
 - `twine check dist/*`
 - `twine upload dist/*` and fill credentials
   - (`twine upload --repository=testpypi dist/*` to publish to [test.pypi.org](https://test.pypi.org/))

No need to explain what `twine check` or `twine upload` does, I guess the names are self-explanatory 😉 And if someone wants to know more about these commands for whatever reason, all what it takes is to paste them into Google.

Reply:
Yeah, I reformatted this a few times while writing it. The shell code block made more sense without comments in between 🙂

I commented/explained a bit more than usual, because someone following this guide might be less familiar with how Python packaging works - when you have to publish runtimes for 5 different languages, there will always be some languages that you know less well than others 🙂 But yeah, the commands have relatively descriptive names, so some of these explanations probably aren't necessary.